### DISCUSSION TITLE
No backup files

### DISCUSSION THREAD

**Discussion by TBrown899** [freeplane/freeplane#1675] [Help]
I'm making a last ditch effort to find a backup file -- although I expect some hours of work over the past few days is lost because, while my Freeplane preferences are set to make backup files, it seems they are not being made. A file I was working on... is somehow back to an old version... I guess because I didn't hit save more recently however I seem to recall hitting save... I did something which I'm not clear about... Anyway. Gone, gone, gone? And why if the autosave is on with file location specified, files aren't appearing there? Thanks,

**Comment by dpolivaev** [comment]
The general but useless answer is that it is because we do not live in an ideal world.

Autosave helps only if Freeplane crashes. If it terminates properly, the files are deleted on exit.
If save does not work even when you press "save" there is no reason why autosave should behave better.
Generally, if you execute File->Reload mind map version, it shows you all available copies of the map with their dates.
The backup directory is configured at Preferences…->Environment->Automatic save->Backup directory (if above option is selected) and I only know it because I know how to use command search dialog which allows me to quickly find all menu actions and preferences.

Regards.

**Comment by euu2021** [comment]
>  while my Freeplane preferences are set to make backup files, it seems they are not being made

Is your backup folder empty?

I just checked here that my backups go back to January 21.

**Comment by TBrown899** [comment]
Before I continue I must give a big thank you to the Freeplane team both for the app and support. I first started using the app in 2011 and perhaps earlier. After a long break of regular use, I'm very impressed with the iconic interface improvements with my recent install update.

Back to saving files. This is scary! I very much don't understand yesterday's feedback. Anyway. What may have happened is that I had an instance of Freeplane open with an early version of my file, had closed other instances, and then blindly concluded by closing the old instance so overwriting updates. Anyway. I was expecting to find backups somewhere to save me but there are none. Yes. Backup folder is empty though the preferences are set to backup and specify a folder. Now, for urgent requirements, I need to understand basic file saving here -- before I lose other recent work. I will go look at files now...

Thanks,

**Comment by TBrown899** [comment]
As I assess my current working files, I notice that some are showing autosave files when I view Reload mind map version while others are not showing a selection of autosave files. Now, I will search my drive from *.autosave files... Okay, I need to search for where Freeplane is saving these files as they don't appear on drive search...

**Comment by TBrown899** [comment]
I'm not seeing the AppData\Roaming\Freeplane\1.11x folder where I expect to find it, as specified in preferences, automatic save (where boxes are checked)... I will search for how to show hidden folders....

**Reply by TBrown899** [reply]
Okay, I'm now showing hidden system files and in the freeplane folder, it shows autosave files for my target document however only for my most recent revisions yesterday, revisions to the old file from 3 days ago, nothing for the more mature contents.

**Reply by TBrown899** [reply]
So, it looks like a lost cause. I will simply reconstruct the map from scratch... and check here that the app is creating backups as I move forward....

**Reply by TBrown899** [reply]
I host my freeplane working files in Dropbox.

**Reply by TBrown899** [reply]
OMG. I found it. It was there in AppsData. Yeah!

**Comment by euu2021** [comment]
For more backup ideas, take a look at this discussion: https://github.com/freeplane/freeplane/discussions/1290

The app mentioned there, created by macmarrum, is working very well for me, using those settings: https://github.com/macmarrum/rumar/issues/2#issuecomment-1638777416

Take a look how there is a good coverage of the last months, without many files:
![image](https://github.com/freeplane/freeplane/assets/77707706/d627def8-0771-4fac-bead-5336dca34c1c)

And there is also the vanilla Freeplane backup, that covers the more recent days.


And it's relevant to note that you need to manually backup your media folder, because the automatic backups don't include it.

**Reply by TBrown899** [reply]
Is there a problem with Freeplane backup? 
I resolved my issue so I was thinking it's just me. 
Do I actually need to take measures to ensure consistent backups? The app has some issues there?
Thanks,

**Reply by euu2021** [reply]
There is no problem with the Freeplane backup, but the user may want to have more control over the frequency that the backups are created. For example, I like to keep one backup for each month in the last year; 1 backup for each week of the last couple of weeks; 1 backup for each day of the last couple of days.

**Comment by TBrown899** [comment]
Thank you for the link to more information on backups. (I need to get this down!)
